components/user-mgt/org.wso2.carbon.user.mgt/src/test/java/org/wso2/carbon/user/mgt/UserRealmProxyTest.java

```diff
@@ -0,0 +1,277 @@
+        //  HybridRoleManager hybridRoleManager = mock(HybridRoleManager.class);
+        Field f1 = cc.getClass().getSuperclass().getDeclaredField("hybridRoleManager");
+        f1.setAccessible(true);
+        f1.set(cc, hybridRoleManager);
+
+        String[] test = {"role3x", "role4x"};
+        Mockito.when(hybridRoleManager.getHybridRoles("role")).thenReturn(test);
+        Mockito.when(((AbstractUserStoreManager) userStoreManagerWithAb).getSharedRoleNames("role", 10)).thenReturn(new String[]{"role1", "role2"});
+        Mockito.when(((AbstractUserStoreManager) userStoreManagerWithAb).getRoleNames("role",
+                10, true, true, true)).thenReturn(null);
+        FlaggedName[] roleList = userRealmProxy.getAllRolesNames("role", 10);
+        Assert.assertEquals(roleList.length, 3);
+    }
+
+    @Test
+    public void testGetUserRealmInfo() throws Exception {
+        Mockito.when(realm.getAuthorizationManager()).thenReturn(authorizationManager);
+        Mockito.when(realm.getClaimManager()).thenReturn(claimManager);
+        Mockito.when(authorizationManager.isUserAuthorized("admin",
+                "/permission/admin/manage/identity", CarbonConstants.UI_PERMISSION_ACTION)).thenReturn(true);
+
+
+        Mockito.when(realm.getRealmConfiguration()).thenReturn(this.getSampleRelaimConfiguration());
+        Mockito.when(realm.getUserStoreManager()).thenReturn(userStoreManagerWithAb);
+        ClaimMapping claimMapping = new ClaimMapping();
+        claimMapping.setMappedAttribute("test1");
+        claimMapping.setMappedAttribute("test2");
+        claimMapping.setMappedAttribute("test3");
+
+
+        Claim claim = new Claim();
+        claim.setClaimUri("testURI");
+        claim.setValue("testClaim");
+        claimMapping.setClaim(claim);
+
+        Mockito.when(claimManager.getAllClaimMappings(UserCoreConstants.DEFAULT_CARBON_DIALECT)).thenReturn(new ClaimMapping[]{claimMapping});
+        Mockito.when(userStoreManagerWithAb.getRealmConfiguration()).thenReturn(this.getSampleRelaimConfiguration());
+        Mockito.when(userStoreManagerWithAb.isBulkImportSupported()).thenReturn(false);
+        UserRealmInfo realmInfo = userRealmProxy.getUserRealmInfo();
+        Assert.assertEquals(realmInfo.getEveryOneRole(), "everyone");
+        Assert.assertEquals(realmInfo.getAdminRole(), "admin");
+
+    }
+
+    @Test
+    public void testAddUser() throws Exception {
+        Mockito.when(realm.getRealmConfiguration()).thenReturn(this.getSampleRelaimConfiguration());
+        Mockito.when(realm.getUserStoreManager()).thenReturn(userStoreManager);
+        Mockito.when(realm.getAuthorizationManager()).thenReturn(authorizationManager);
+        Mockito.when(authorizationManager.isRoleAuthorized("role1", PERMISSION, UserMgtConstants.EXECUTE_ACTION)).thenReturn(true);
+        userRealmProxy.addUser("testUser", "password", new String[]{"role1", "role2"}, getSampleClaims(), "default");
+        Assert.assertTrue(true);
```

Comment:
Isn't it better to use ArgumentCaptor to capture an argument inside the method and use that for the assert? :)

Reply:
This is OK since we can see where the Assert came from in some platforms there are several assert types from different interfaces